### ISSUE TITLE
Authentication Return calls before initialization done

### ISSUE THREAD

**Issue by kenglou** [playgameservices/play-games-plugin-for-unity#375]
If i do init game services first and straight away continue with Social.Active.Authenticate, I'll directly receive a return false as I can still choose which account i want to login. 

This happens only after i had login GooglePlayServices before and I reinstall the game.

The problem I found is the AuthState haven't changed to silent pending when I'm calling Authenticate functions. To solve the problem i create an init state for the AuthState and reject any callback when Player is in This State.

**Comment by KrystianN** [comment]
I have the same issue.

> To solve the problem i create an init state for the AuthState and reject any callback when Player is in This State.

Could you elaborate a little bit more? How did you accomplished this in your code?

**Comment by ImranZahid** [comment]
I am having similar issue. I have a login button inside my game and it works fine. Only problem is when I signOut all accounts from Settings->Accounts section. I run my game and inside I click on LoginButton, it asks me to enter account details and shows me I signed in successfully but it returns 'false' as Social.localUser.Authenticate call back. The next time if I click its fine, only happens once at start.

**Comment by Laikkae** [comment]
I also had the same issue with 0.9.12 and 0.9.15 (didn't try with older versions). After some testing I found out that the authentication process fires incorrectly before initialization is done **only** if saved games are enabled:

```c#
PlayGamesClientConfiguration config = new PlayGamesClientConfiguration.Builder()
    .EnableSavedGames()
    .Build();
PlayGamesPlatform.InitializeInstance(config);
PlayGamesPlatform.DebugLogEnabled = true;
PlayGamesPlatform.Activate();
```

Noteworthy is that the correct authentication process starts normally, after the initialization is done, but the callback is already fired by the process that started too early. So basically authentication transition is handled twice (HandleAuthTransition() method in NativeClient.cs). And as said before, this happens only after launching the game first time after one has authenticated before and then re-installed the game.

**Comment by claywilkinson** [comment]
@Laikkae  thanks for the details - we'll make sure to address this soon.

**Comment by CodingStudios** [comment]
I added a note to issue #443. Changing the callback to an actual method rather than a bool success in has resolved my issue with authentication. I have tested this with saved games and without.

**Comment by claywilkinson** [comment]
This is resolved in version 0.9.20 of the plugin.  Please re-open if there are still questions.

**Comment by ingridelin** [comment]
I really need this fix, but can't update since we're due to release in a few days. Is there any way I can get a patch to fix this? I'm on version 0.9.11 of the plugin.
